### DISCUSSION TITLE
Export using Dynamic Component variable

### DISCUSSION THREAD

**Discussion by synelle** [lairdubois/lairdubois-opencutlist-sketchup-extension#431] [Ideas]
Hello everyone,

For the new export function, I thought that the descritption was the Dynamic Component description variable, but it seems that it's specific to OCL.

Since I have a lot of auto-generated text, is there a way to link a DC variable to a column in OCL ?
 
![OPC](https://user-images.githubusercontent.com/3049716/193796382-e31cc5ff-e1d1-4f92-ab8f-e963deadd2d2.png)

**Comment by bbeaulant** [comment]
Hi,

Description is not an OpenCutList specific field. It's the native **description** defined on the **component definition**.
Maybe the DC description is an other thing ... 🤔

<img width="1467" alt="Capture d’écran 2022-10-04 à 15 42 42" src="https://user-images.githubusercontent.com/156056/193835545-1a39539e-e0a3-43f9-81b9-544a08408e1f.png">

**Reply by bbeaulant** [reply]
The main problem with DC attributes is that they break the component definition concept. Because their values are computed on instance side.
Then a unique component definition could present different DC attributes values. But OpenCutList groups parts by component definition. Then all grouped part need to share the exact same "attributes".

A similar problem was solve by using the DC "Name" attribute instead of the component definition name. But in this case different name = different part. And if over that we add each DC attribute, it will be to complex ...

<img width="1689" alt="Capture d’écran 2022-10-04 à 16 04 24" src="https://user-images.githubusercontent.com/156056/193840703-f77814f1-f312-49b0-ad53-8eda7ce66ccb.png">
